vizro-core/src/vizro/models/_action/_action.py

```diff
@@ -60,32 +60,82 @@ def _runtime_args(self) -> dict[str, _IdProperty]:
     def _action_name(self) -> str:
         raise NotImplementedError
 
+    def _get_component_id_and_property(self, reference: str, *, type: Literal["input", "output"]) -> tuple[str, str]:
```

Comment:
Generally agree with the logic here but I have a slightly reworking of it which will also make it easier to integrate @petar-qb's forthcoming changes - will share tomorrow.